core/services/editor/BlockAssetRegister.php

```diff
@@ -0,0 +1,219 @@
+    private $registry;
+
+    /**
+     * @var string $script_handle
+     */
+    private $script_handle;
+
+    /**
+     * @var array $script_dependencies
+     */
+    private $script_dependencies;
+
+    /**
+     * @var string $style_handle
+     */
+    private $style_handle;
+
+    /**
+     * @var array $style_dependencies
+     */
+    private $style_dependencies;
+
+
+    /**
+     * BlockAssetRegister constructor.
+     *
+     * @param string          $script_handle
+     * @param array           $script_dependencies
+     * @param string          $style_handle
+     * @param array           $style_dependencies
+     * @param DomainInterface $domain
+     * @param Registry        $registry
+     * @throws InvalidDataTypeException
+     */
+    public function __construct(
+        $script_handle,
+        array $script_dependencies,
+        $style_handle,
+        array $style_dependencies,
+        DomainInterface $domain,
+        Registry $registry
+    ) {
+        $this->setScriptHandle($script_handle);
+        $this->setScriptDependencies($script_dependencies);
+        $this->setStyleHandle($style_handle);
+        $this->setStyleDependencies($style_dependencies);
+        $this->domain              = $domain;
+        $this->registry            = $registry;
+    }
+
+
+    /**
+     * @return string
+     */
+    public function scriptHandle()
+    {
+        return $this->script_handle;
+    }
+
+
+    /**
+     * @param string $script_handle
+     * @throws InvalidDataTypeException
+     */
+    private function setScriptHandle($script_handle)
+    {
+        if (! is_string($script_handle)) {
+            throw new InvalidDataTypeException('$script_handle', $script_handle, 'string');
+        }
+        $this->script_handle = $script_handle;
+    }
+
+
+    /**
+     * @return string
+     */
+    public function styleHandle()
+    {
+        return $this->style_handle;
+    }
+
+
+    /**
+     * @param string $style_handle
+     * @throws InvalidDataTypeException
+     */
+    private function setStyleHandle($style_handle)
+    {
+        if (! is_string($style_handle)) {
+            throw new InvalidDataTypeException('$style_handle', $style_handle, 'string');
+        }
+        $this->style_handle = $style_handle;
+    }
+
+
+    /**
+     * @return array
+     */
+    private function scriptDependencies()
+    {
+        return $this->script_dependencies;
+    }
+
+
+    /**
+     * @param array $script_dependencies
+     */
+    private function setScriptDependencies(array $script_dependencies)
+    {
+        $this->script_dependencies = $script_dependencies + array(
+                'eejs-core',
+                'wp-blocks',    // Provides useful functions and components for extending the editor
+                'wp-i18n',      // Provides localization functions
+                'wp-element',   // Provides React.Component
+                'wp-components' // Provides many prebuilt components and controls
```

Comment:
hmm.  I don't know if we should automatically assume all these dependencies would be registered with every script.  Certainly in the editor context it will usually be true, but in the frontend context it may not be.